test/expect.js

```diff
@@ -1462,6 +1462,18 @@ describe('expect', function () {
     err(function () {
       expect(undefined, 'blah').to.have.property("a");
     }, "blah: Target cannot be null or undefined.");
+
+    expect(function () {
+      expect({a:1}).to.have.nested.property('{a:1}');
+    }).to.not.throw('the argument to `property` must be a string');
+
+    expect(function () {
+      expect({a:1}).to.have.nested.property({'a':'1'});
+    }).to.throw('the argument to `property` must be a string');
+
+    expect(function () {
+      expect({a:1}).to.have.property(null);
+    }).to.throw('the argument to `property` must be either of type string, number or symbol');
```

Comment:
Let's repeat these three new error tests in the `should` and `assert` interface test files.

Reply:
Okay , so just duplicating them in both the files.

Reply:
@keithamus @meeber `isNested` is true when `name` is `undefined` , Shouldn't this be false

Reply:
@solodynamo I'm not sure what you mean.

On a separate note, I just noticed a couple of things:

- Let's remove the `.not.throw` test on lines 1466-1468; we already test successful `.nested.property` assertions in the section that starts on line 1754.
- Let's relocate the test on lines 1470-1472 to the end of the `nested.property` section (that starts on line 1754).

Reply:
@meeber I meant to ask 
` expect(function () {
      ({a:1}).should.have.nested.property(undefined);
    }).to.throw('the argument to `property` must be either of type string, number or symbol');` 
Above test is throwing error `'the argument to `property` must be a string'`. I debugged and found `isNested` is true in above case , so i meant to ask shouldn't `isNested` be false. Please correct me if i misunderstood something.